actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -211,7 +211,7 @@ def request_http_digest_authentication(realm = "Application", message = nil)
         end
       end
 
-      # Returns false on a valid response, true otherwise.
+      # Returns true on a valid response, nil otherwise.
       def authenticate(request, realm, &password_procedure)
         request.authorization && validate_digest_response(request, realm, &password_procedure)
```

Comment:
`authenticate` can return false as well, as `validate_digest_response` can return false.
So to keep things consistent I think we should always return true or false by using `present?`.

```suggestion
      # Returns true on a valid response, false otherwise.
      def authenticate(request, realm, &password_procedure)
        request.authorization.present? && validate_digest_response(request, realm, &password_procedure)
```

This is also inline with this class as we also check presence on `request.authorization` in the same class here:
https://github.com/rails/rails/blob/5bec50bc70380bb1e70e8fb0a1654130042b1f16/actionpack/lib/action_controller/metal/http_authentication.rb#L115

Reply:
@p8 I have made the changes.

Reply:
@p8 Please check.

Reply:
@p8 When will this PR be merged?